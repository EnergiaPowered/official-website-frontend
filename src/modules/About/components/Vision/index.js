import React from "react";

/**
 * Component which contain The Vision of Energia Powered
 * 
 * @component 
 * @returns {JSX} Return the vision that passed
 */

export default ({ vision }) => {
  return (
    <section className="bg-section about-vision component-font">
      <h2 className="section-title">Our Vision</h2>
      <p data-testid="vision-text"> {vision} </p>
    </section>
  );
};
