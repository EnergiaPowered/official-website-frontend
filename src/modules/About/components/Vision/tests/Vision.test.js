import React from "react";

import Vision from "../index";

import { create } from "react-test-renderer";
import { cleanup, render } from "@testing-library/react";

import '@testing-library/jest-dom/extend-expect'

afterEach(cleanup)

// matches snapshot
it("matches snapshot", () => {
  const component = create(<Vision />);

  expect(component.toJSON()).toMatchSnapshot();
});

// renders with correct props
it("renders with the correct props", () => {
    const { getByTestId } = render(<Vision vision="Energia Powered" />)
    expect(getByTestId("vision-text")).toHaveTextContent("Energia Powered")
})
