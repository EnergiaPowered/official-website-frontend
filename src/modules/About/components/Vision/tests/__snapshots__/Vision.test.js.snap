// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="bg-section about-vision component-font"
>
  <h2
    className="section-title"
  >
    Our Vision
  </h2>
  <p
    data-testid="vision-text"
  >
     
     
  </p>
</section>
`;
